apps/spruce/src/pages/waterfall/WaterfallMenu/GitCommitSearch.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import TextInput from "@leafygreen-ui/text-input";
+import { Description } from "@leafygreen-ui/typography";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useWaterfallAnalytics } from "analytics/waterfall/useWaterfallAnalytics";
+import { DropdownItem } from "components/ButtonDropdown";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { useQueryParams } from "hooks/useQueryParam";
+import { WaterfallFilterOptions } from "../types";
+
+interface GitCommitSearchProps {
+  setMenuOpen: (open: boolean) => void;
+}
+
+export const GitCommitSearch: React.FC<GitCommitSearchProps> = ({
+  setMenuOpen,
+}) => {
+  const { sendEvent } = useWaterfallAnalytics();
+  const [, setQueryParams] = useQueryParams();
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [commitHash, setCommitHash] = useState("");
+
+  const onCancel = () => {
+    setModalOpen(false);
+    setMenuOpen(false);
+  };
+
+  const onConfirm = () => {
+    sendEvent({ name: "Filtered by git commit" });
+    setQueryParams({
+      [WaterfallFilterOptions.Revision]: commitHash,
+    });
+    onCancel();
+  };
+
+  return (
+    <>
+      <DropdownItem
+        data-cy="git-commit-search"
+        onClick={() => {
+          setModalOpen(true);
+          sendEvent({ name: "Viewed git commit modal" });
```

Comment:
This seems only marginally useful to me; I guess it would allow us to see how many users looked at the modal without actually submitting a git hash query but that doesn't seem very insightful. I wouldn't mind dropping this if you didn't have a strong reason for it.